src/librustc_driver/driver.rs

```diff
@@ -574,19 +546,28 @@ fn count_nodes(krate: &ast::Crate) -> usize {
 // For continuing compilation after a parsed crate has been
 // modified
 
+pub struct Phase2Results<'a> {
```

Comment:
Could this have a more descriptive name? PostExpansionResults? or ExpansionResult or ExpandedCrate, etc.